build/index.js

```diff
@@ -35,124 +42,109 @@ const bootstrap = () => {
     if (!fs.existsSync(elmJsonFile)) {
       generateElmJson();
     }
-    return Promise.resolve();
   } catch (e) {
-    return Promise.reject(`Could not complete bootstrap: ${e}`);
+    return new Error(`Could not complete bootstrap: ${e}`);
   }
 };
 
-const fetchIconsArchive = () => {
-  try {
-    const archiveName = "zondicons.zip";
-    const srcArchivePath = `${__dirname}/${archiveName}`;
-    const dstArchivePath = `${tmpPath}/${archiveName}`;
+const wait = (duration) => (...args) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(...args);
+    }, duration);
+  });
+};
 
-    fs.copyFileSync(srcArchivePath, dstArchivePath);
+const fetchIcons = async () => {
+  const stream = require("stream");
+  const { promisify } = require("util");
+  const fetch = require("node-fetch");
+  const unzipper = require("unzipper");
+  const iconsUrl = "https://www.zondicons.com/zondicons.zip";
+  const asyncStreamPipeline = promisify(stream.pipeline);
 
-    return Promise.resolve(dstArchivePath);
-  } catch (e) {
-    return Promise.reject(`Could not fetch icons archive: ${e}`);
-  }
+  const res = await fetch(iconsUrl);
+
+  return asyncStreamPipeline(res.body, unzipper.Extract({ path: tmpPath }))
+    .then(wait(250))
+    .then(() => `${tmpPath}/${basename(iconsUrl)}`);
 };
 
 const basename = (fileName) => path.basename(fileName, path.extname(fileName));
 
-const extractIcons = (archive) => {
-  if (!fs.existsSync(archive)) {
-    return Promise.reject("Could not find icons archive");
-  }
-  try {
-    const zip = new AdmZip(archive);
-    const extractionPath = `${tmpPath}/icons`;
-    zip.extractAllTo(extractionPath, true);
-    return Promise.resolve(`${extractionPath}/${basename(archive)}`);
-  } catch (e) {
-    return Promise.reject(`Could not extract icons: ${e}`);
-  }
+const createIconName = (name) => {
+  return require("camelcase")(name);
 };
 
-const createModuleName = (name) => {
-  return require("pascal-case").pascalCase(name);
+const createElmContent = ({ iconName, body, funcNames }) => {
+  return funcNames.reduce((acc, fn) => {
+    const ret = acc.replace(
+      new RegExp(`${fn}(:| )`, "gm"),
```

Comment:
what does the `m` modifyer do? Would it make sense to match from the start of the line here with `^`

Reply:
m is for multi line match. Matching a line start would change nothing in this case

Reply:
Ah ok